content/docs/user-guide/experiment-management/running-experiments.md

```diff
@@ -0,0 +1,569 @@
+# Running Experiments
+
+## Motivation
+
+Machine Learning and Data Science projects usually involve experimentation.
+Motivation of these experiments can range from finding good hyperparameters to
+checking data and concept drift. DVC 2 introduced a new set of commands to
+manage experiments with minimum boilerplate. It allows to run experiments
+defined by pipelines, track their associated data and model files, set
+parameters for each, push experiment parameters and code to Git remotes without
+committing them, create branches and persist them.
+
+In this part of the DVC User's Guide we will explain how DVC runs the
+experiments, setting parameters, running the experiments in parallel, or in
+temporary directories.
+
+If this is the first time you're introduced to Machine Learning experimentation,
+it may be better to start with the
+[Experiment's Trail](/doc/start/experiments/).
+
+## The Pipeline
+
+DVC uses pipelines composed of <abbr>stages</abbr>. Stages are isolated and
+interdependent steps that has a command, a set of dependencies, and output.
+Stages can depend onto each other and can build a <abbr>DAG</abbr>.
+
+DVC employs the pipeline to run the experiment commands. Each pipeline has an
+end-point, which is the last command in the pipeline. The output of this last
+command is considered the output of experiment. Additionally DVC tracks
+artifacts from intermediate stages, and doesn't reproduce them if their
+dependents have not changed.
+
+DVC pipelines are a detailed topic that we cover in [Get Started][gs-pipelines]
+and the [User's Guide][ug-pipeline-files]. Here we assume that there is already
+a pipeline defined in `dvc.yaml` file in the <abbr>project</abbr>.
+
+[gs-pipelines]: /doc/start/data-pipelines
+[ug-pipeline-files]: /doc/user-guide/project-structure/pipelines-files
+
+### Running the pipeline
+
+The pipeline defined in `dvc.yaml` file is run with default settings using:
+
+```dvc
+$ dvc exp run
+```
+
+If all dependencies and the experiment output are not changed and available in
+the workspace, the `dvc exp run` doesn't rerun the commands.
+
+For a pipeline composed of `extract`, `transform`, `train`, `evaluate`, if a
+dependency of `train` stage has changed, the dependent stages (`evaluate`) are
+also run.
+
+### Specifying Targets
+
+By default `dvc exp run` uses `dvc.yaml` file in the current directory. You can
```

Comment:
Actually it should use all the dvc.yaml files in the repo by default. See https://dvc.org/doc/command-reference/repro#options (BTW this may be wrong in the help output, if so feel free to open a core repo issue or contribute a change).

Reply:
p.s. I'm not 100% sure about this.

Reply:
BTW since we'll have a guide should we remove many of the details from that cmd ref **Options** section I linked to? `targets` is not even an option (flag), but rather a (required) argument.

In fact this idea of removing details from the cmd ref can be generalized but Idk if we want to do as part of this same PR (maybe after some more review iterations) or in a different one. WDYT?

Reply:
As we progress to develop the guide, we can remove the repeating sections in the command reference. Command samples with a single sentence may remain in the cmd ref, but longer discussions around examples should be in the UG I think.

Reply:
I tested `dvc exp run`, `dvc exp run --all-pipelines`, `dvc exp run dir/dvc.yaml` and the first of these doesn't run the pipeline in another directory. Even `dvc exp run -R .` doesn't run.

Reply:
I think `dvc exp run` has no feature parity with `dvc repro` when it comes to recursive directories or multiple `dvc.yaml`s.

Reply:
I think I was wrong about the default target being all dvc.yaml files. I know that repro does check all the dvc.yaml files in the repo for consistency though, but only executes pipelines included in the targets.

> As we progress to develop the guide, we can remove the repeating sections in the command reference

Agree, that's what I meant. My question is when. I think if the changes are not major it could be included in this same PR so we're done with it. But a separate one is also OK (add a checkbox?)

> dvc exp run has no feature parity with dvc repro when it comes to recursive directories or multiple dvc.yamls

`exp run` should accept all the same arguments and options as `repro` and behave the same way. See https://dvc.org/doc/command-reference/exp/run#options

Reply:
> exp run should accept all the same arguments and options as repro and behave the same way. See https://dvc.org/doc/command-reference/exp/run#options

Yes, _should_ :) 

Before the bug reports (iterative/dvc#6456 and iterative/dvc#6457), I had no strong opinion on this but from the development and future directions, I think it's better to separate `dvc exp run` from `dvc repro` in documentation. Some elements of `repro` may be obsolete in `exp run`, or vice versa. Considering these as two distinct commands seems more prudent to me.

Reply:
> a separate one is also OK (add a checkbox?)

Added a checkbox to the description of this PR for now. Resolving

> it's better to separate dvc exp run from dvc repro in documentation. Some elements of repro may be obsolete in exp run.

Let's wait until that's an official decision of the DVC product team.